Core/include/Acts/Navigation/NavigationHelpers.hpp

```diff
@@ -0,0 +1,69 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Detector/Portal.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Navigation/NavigationState.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+
+namespace Acts {
+namespace Experimental {
+
+/// Helper method to update the candidates (portals/surfaces),
+/// this can be called for initial surface/portal estimation,
+/// but also during the navigation to update the current list
+/// of candidates.
+///
+/// @param gctx is the Geometry context of this call
+/// @param nState [in,out] is the navigation state to be updated
+///
+/// @todo for surfaces skip the non-reached ones, while keep for portals
+inline static void updateCandidates(
```

Comment:
Do these need to be inline?

Reply:
I guess not - I just saw that removing that would lead to linker errors in some cases but maybe that was also in combination with templates. the question is also if these kind of methods really have to have their impl in the header